lightcurve/views.py

```diff
@@ -175,3 +177,26 @@ def all_lightcurve(request):
 
     return render_to_response('lightcurve_list.html', {"contexts": context_list, "request_all": True},
             context_instance=RequestContext(request))
+
+
+def download(request, lightcurve_id):
+    file_type = request.GET.get('file_type')
+    response = HttpResponse(content_type='text/csv')
+    response['Content-Disposition'] = 'attachment; filename="LightCurve' + lightcurve_id + '.' + file_type + '"'
+
+    lc = LightCurve.objects.get(id=lightcurve_id)
+    analyses = ImageAnalysis.objects.filter(useruploadedimage__lightcurve=lc) \
+                                    .exclude(status=ImageAnalysis.ASTROMETRY_PENDING)
+
+    if file_type == "csv":
+        writer = csv.writer(response)
+        writer.writerow(['Datetime', 'JD', 'Mag instrumental', 'Mag standard', 'Mag std'])
+        for analysis in analyses:
+            if analysis.annotated_point_sources != "[]":
```

Comment:
Are you sure this shouldn't be `if analysis.annotated_points_sources != []`?  If you are seeing a string containing `"[]"`, that might be a problem with the database I sent you.

`if analysis.annotated_point_sources` would also be equivalent

Reply:
Tried both ([] and just `if analysis.annotated_point_sources`). Both gets `if point['id'] == pid:
TypeError: string indices must be integers` error message. I am not sure why?